File: docs/snippets/all/archetypes/encoded_image.cpp
// Create and log a image.

#include <dalaran.hpp>

#include <filesystem>
#include <fstream>
#include <iostream>
#include <vector>

namespace fs = std::filesystem;

int main(int argc, char* argv[]) {
    const auto rec = dalaran::RecordingStream("dalaran_example_encoded_image");
    rec.spawn().exit_on_failure();

    fs::path image_filepath = fs::path(__FILE__).parent_path() / "ferris.png";

    rec.log(
        "image",
        dalaran::EncodedImage::from_file(image_filepath).value_or_throw()
    );
}

File: docs/snippets/all/archetypes/encoded_image.py
"""Create and log an image."""

from pathlib import Path

import dalaran as dl

image_file_path = Path(__file__).parent / "ferris.png"

dl.init("dalaran_example_encoded_image", spawn=True)

dl.log("image", dl.EncodedImage(path=image_file_path))
